pkgs/build-support/libredirect/default.nix

```diff
@@ -5,15 +5,14 @@ stdenv.mkDerivation {
 
   unpackPhase = "cp ${./libredirect.c} libredirect.c";
 
-  buildPhase =
-    ''
-      gcc -Wall -std=c99 -O3 -shared libredirect.c -o libredirect.so -fPIC -ldl
-    '';
+  buildPhase = ''
+    $CC -Wall -std=c99 -O3 -shared libredirect.c -o libredirect.so -fPIC -ldl
```

Comment:
technically speaking the file extension should be .dylib on Darwin. I don't know if it's worth changing at this change will force all the users to also add some conditions

Reply:
We could use `stdenv.targetPlatform.extensions.sharedLibrary` here and at runtime we could use `stdenv.hostPlatform.extensions.sharedLibrary`. That way we at least don't need conditions.